index.es.js

```diff
@@ -1,12 +1,5 @@
-export {
-  Future,
-  Future as default,
-  isFuture,
-  reject,
-  of,
-  never,
-  isNever
-} from './src/core';
+export {default, default as Future} from './src/future';
```

Comment:
Right. Maybe you should do `import Future from './src/future'; export default Future;` to be safe.